app/code/Magento/Csp/Helper/InlineUtil.php

```diff
@@ -196,23 +193,8 @@ public function processTag(TagData $tagData): TagData
                     new FetchPolicy($policyId, false, $remotes)
                 );
             }
-            if ($tagData->getContent()
-                && !empty(self::$tagMeta[$tagData->getTag()]['hash'])
-                && $this->isInlineDisabled(self::$tagMeta[$tagData->getTag()]['id'])
-            ) {
-                $this->dynamicCollector->add(
-                    new FetchPolicy(
-                        $policyId,
-                        false,
-                        [],
-                        [],
-                        false,
-                        false,
-                        false,
-                        [],
-                        $this->generateHashValue($tagData->getContent())
-                    )
-                );
+            if (!empty(self::$tagMeta[$tagData->getTag()]['hash'])) {
```

Comment:
rather than adding the sanitisation logic in the main InlineUtil helper, I have chosen to simplify this file and instead delegate the inline whitelisting logic to new model. I was unsure to use models or helpers but the PR as it is does reduce the complexity of this helper file and I hope it is a good way forward